bbc/simorgh
# Set `x-frame-options` in Next.js app

Overall changes
======
- Sets the `x-frame-options` response header to `DENY` across the Next.js app by default
- This header is removed from `av-embeds` routes to allow embedding of the content in iframes

Code changes
======
- Updates `next.config.js` to set the `x-frame-options` header across all page types
- Removes the `x-frame-options` header on `av-embeds` routes

Testing
======
1. Visit http://localhost:7081/burmese/live/cx8jgp1zxylt?renderer_env=live
2. Open Chrome dev tools Network tab and inspect the `.html` page
3. Observe the `x-frame-options: DENY` header in the 'Response Headers' section
4. Visit http://localhost:7081/mundo/send/u50853489?renderer_env=live
5. Open Chrome dev tools Network tab and inspect the `.html` page
6. Observe the `x-frame-options: DENY` header in the 'Response Headers' section
7. Visit http://localhost:7081/serbian/cyr/av-embeds/srbija-50103048/pid/p07rfhrv?renderer_env=live
8. Open Chrome dev tools Network tab and inspect the `.html` page
9. Observe the `x-frame-options` header is not visible in the 'Response Headers' section
10. Visit http://localhost:7081/ws/av-embeds/articles/cd1rmn075d1o/p0jd37n8/ig?renderer_env=live
11. Open Chrome dev tools Network tab and inspect the `.html` page
12. Observe the `x-frame-options` header is not visible in the 'Response Headers' section

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
